### PR TITLE
Day03 golang

### PR DESCRIPTION
**Pull request by ilyasabdellaoui**
- Part 1: pretty straightforward, nothing is tricky, u only need to extract the right mul expressions, then u extract numbers in those expression and do ur ultiplications
- Part 2: the tricky part is to save state weather mults are enabled or disabled based on the last do() or don't(), using an alternation match

### CODE REVIEW

**Comment by ilyasabdellaoui** [comment]
@jamali005 @SaidNajim Review PR #6 before this one, even if this one includes code from PR #6 (realised i didnt create a branch from main for day 3 but rather from day02 branch, well i get it it's 3:00AM)

**Review by SaidNajim — APPROVED** [review]

File: golang/day03/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            doNb,_ := strconv.Atoi(doNbr)
+            doMult *= int(doNb)
+        }
+        doMultsSum += doMult
+    }
+    fmt.Println("Part 2 solution : ", doMultsSum)
+}
+
+func extractMul(input string) []string {
+    re := regexp.MustCompile(`mul\(\d{1,3},\d{1,3}\)`)
```
Comment (jamali005) [inline golang/day03/main.go:49]: if the number more than 3 digits 👀 ?

File: golang/day03/main.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    matches := re.FindAllString(input, -1)
+    return matches
+}
+
+func extractFactors(match string) []string {
+    re := regexp.MustCompile(`\d{1,3}`)
```
Comment (jamali005) [inline golang/day03/main.go:55]: same as previous comment

File: golang/day03/main.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    nbrs := re.FindAllString(match, -1)
+    return nbrs
+}
+
+func extrcatDoMuls(input string) string {
+  	re := regexp.MustCompile(`do\(\)|don't\(\)|mul\(\d{1,3},\d{1,3}\)`)
```
Comment (jamali005) [inline golang/day03/main.go:61]: see above

File: golang/day03/main.go (hunk 1 above)
Comment (ilyasabdellaoui) [reply golang/day03/main.go:49]: @jamali005 i hardcoded it to 3 digs just to comply with the problem statement, i could've replaced `{1,3}` with `+` that matches a number with 1 digit and more

**Review by jamali005 — APPROVED** [review]
Resolved

**Pull request merged**